site/public/ts/shared/views/CircuitView.ts

```diff
@@ -15,18 +15,16 @@ export abstract class CircuitView {
     protected renderer: Renderer;
     protected camera: Camera;
 
-    public constructor(canvasId: string, vw: number = 1, vh: number = 1) {
-        const canvas = document.getElementById(canvasId);
-        if (!(canvas instanceof HTMLCanvasElement))
-            throw new Error("Canvas element not found!");
+    public constructor(canvas: HTMLCanvasElement, vw: number = 1, vh: number = 1) {
         this.canvas = canvas;
         this.renderer = new Renderer(this.canvas, vw, vh);
         this.camera = new Camera(this.canvas.width, this.canvas.height);
 
         this.resize();
     }
 
-    public render(designer: CircuitDesigner, selections: Selectable[], toolManager: ToolManager): void {
+    // toolManager changed to be optional because the exporter doesn't want to create a dummy tool manager
+    public render(designer: CircuitDesigner, selections: Selectable[], toolManager?: ToolManager): void {
```

Comment:
You passed in `undefined` when you called it above, so you should remove the `undefined` parameter passing since it's optional.